Guest post: Final Encoding in RPython Interpreters

MostAwesomeDude opened the pull request:
As discussed on IRC, here's my adventures in transforming a Brainfuck interpreter to use final encoding. I'm enabling edits by maintainers so that y'all can revise and merge it.

If I get to have a thanks, I'd like to thank Armin for the idea, larstiq for pushing me to write it up, and CF Bolz for pointing out a formatting mistake. Also thank you, reviewer.

If I get to have a shoutout, I'd like to mention that rpypkgs is always open to contributions and that you can run this interpreter directly from the command line as `github:rpypkgs/rpypkgs#bf` if you have Nix installed and flakes enabled. Here's Mandelbrot:

    $ nix-prefetch-url https://github.com/MG-K/pypy-tutorial-ko/raw/refs/heads/master/mandel.b
    $ nix run github:rpypkgs/rpypkgs#bf -- /nix/store/ngnphbap9ncvz41d0fkvdh61n7j2bg21-mandel.b

mattip commented:
Take this with a grain of salt, I was not part of the IRC discussion. 

I think some additional context would be nice. You launch right into the interpreter without describing the problem you want to solve or who might be interested to read the post. Also you can mention "to find more uses of rpython, check out https://github.com/rpypkgs/rpypkgs" or so.

mattip commented on posts/2024/11/guest-post-final-encoding-in-rpython.md:
```diff
@@ -0,0 +1,584 @@
+.. author:
+-->
+
+# Guest Post: Final Encoding in RPython Interpreters
```
The title is automatically generated by nikola, see the [preview](https://deploy-preview-146--keen-mestorf-442210.netlify.app/blog/)

```suggestion
```

mattip commented on posts/2024/11/guest-post-final-encoding-in-rpython.md:
```diff
@@ -0,0 +1,584 @@
+.. link: 
+.. description: 
+.. type: text
+.. author:
```
You probably want to give yourself credit in the author tag. Is this correct? Otherwise this is rendered as written by the PyPy team
```suggestion
.. author: MostAwesomeDude
```

mattip reviewed (commented):
Thanks for the blog post!

cfbolz commented:
> If I get to have a thanks, I'd like to thank Armin for the idea, larstiq for pushing me to write it up, and CF Bolz for pointing out a formatting mistake. Also thank you, reviewer.
> 
> If I get to have a shoutout, I'd like to mention that rpypkgs is always open to contributions and that you can run this interpreter directly from the command line as `github:rpypkgs/rpypkgs#bf` if you have Nix installed and flakes enabled. 

Please just add both of these to the post! Both make perfect sense, and we don't have a fixed format for guest blog posts or anything anyway.

MostAwesomeDude commented:
Bump incoming. Thanks for feedback! I addressed everything, added a couple more links, and also fixed the spell-checker.

I'm adding a link to [this cleaned IRC log](https://gist.github.com/MostAwesomeDude/fd86ad2d2e38af7aa67b6e548aabe008) which contains me and Armin; I don't mind sharing this but Armin can always choose to remove it. There really isn't that much context: I was contemplating adding JSON output to another part of my [Cammy](https://esolangs.org/wiki/Cammy) interpreter, had some desires for generalizing that work, and realized that I could concretely test out one architectural idea on a small interpreter.

mattip reviewed (approved):
LGTM, thanks! I will give other reviewers a chance to take a look before merging soon.
